minigalaxy/installer.py

```diff
@@ -167,6 +170,35 @@ def copy_thumbnail(game):
     return error_message
 
 
+def create_applications_file(game):
+    error_message = ""
+    preferences_switch = Config.get("create_applications_file")
+    if game.platform == "linux" and preferences_switch:
+        path_to_shortcut = os.path.join(APPLICATIONS_DIR, game.name+".desktop")
+        # Create desktop file definition
+        desktop_context = {
+            "game_bin_path": os.path.join(game.install_dir, 'start.sh'),
+            "game_name": game.name,
+            "game_icon_path": os.path.join(game.install_dir, 'support/icon.png')
+            }
+        desktop_definition = """\
+            [Desktop Entry]
+            Type=Application
+            Terminal=false
+            StartupNotify=true
+            Exec="{game_bin_path}"
+            Name={game_name}
+            Icon={game_icon_path}""".format(**desktop_context)
+        if not os.path.isfile(path_to_shortcut):
+            try:
+                with open(path_to_shortcut, 'w+') as desktop_file:
+                    desktop_file.writelines(textwrap.dedent(desktop_definition))
+            except Exception as e:
+                os.remove(path_to_shortcut)
```

Comment:
I am not sure about this...
If we were unable to write to file, we will probably crash also on remove file attempt.

Reply:
I'll do some testing with less permissions to verify.

Reply:
I created a brand new user and installed some games. There were no permission issues but I'm OK to remove it from the exception if you think that's fine.

Also, I referenced https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html and some StackOverflow posts and I don't see a way that the user will not have permissions for ~/.local/share